src/pages/match.js

```diff
@@ -1,24 +1,100 @@
 import React from "react";
 import PropTypes from "prop-types";
-import Page from "../components/Page";
+import { getMatchFetchStatus, getMatch } from "../selectors/MatchSelectors";
+import { getEventFetchStatus, getEvent } from "../selectors/EventSelectors";
+import { fetchMatch, fetchEvent } from "../actions";
+import useData from "../lib/useData";
+import notFoundError from "../lib/notFoundError";
+import Grid from "@material-ui/core/Grid";
 import Typography from "@material-ui/core/Typography";
+import Page from "../components/Page";
+import Link from "../components/Link";
+import MatchBreakdownTable from "../components/MatchBreakdownTable";
+import MatchVideos from "../components/MatchVideos";
+
+const Match = ({ matchKey, refetchOnLoad }) => {
+  const [match, matchFetchStatus, refetchMatch] = useData(
+    state => getMatchFetchStatus(state, matchKey),
+    state => getMatch(state, matchKey),
+    React.useMemo(() => fetchMatch(matchKey), [matchKey]),
+    refetchOnLoad.match
+  );
+  const eventKey = matchKey.split("_")[0];
+  const [event, eventFetchStatus, refetchEvent] = useData(
+    state => getEventFetchStatus(state, eventKey),
+    state => getEvent(state, eventKey),
+    React.useMemo(() => fetchEvent(eventKey), [eventKey]),
+    refetchOnLoad.event
+  );
+  const handleRefresh = React.useCallback(() => {
+    refetchMatch();
+    refetchEvent();
+  }, [refetchMatch, refetchEvent]);
+
+  if (!match || !event) {
+    return notFoundError();
+  }
 
-const Match = ({ matchKey }) => (
-  <Page title={`Match ${matchKey}`} metaDescription="TODO">
-    <Typography variant="h4">Match {matchKey}</Typography>
-    <Typography>TODO</Typography>
-  </Page>
-);
+  return (
+    <Page
+      title={`${match.getDisplayName()} - ${event.name} (${event.year})`}
+      metaDescription={`Match results and video for ${match.getDisplayName()} at the the ${
+        event.year
+      } ${
+        event.name
+      } FIRST Robotics Competition in ${event.getCityStateCountry()}.`}
+      isLoading={
+        matchFetchStatus === "fetching" || eventFetchStatus === "fetching"
+      }
+      refreshFunction={handleRefresh}
+    >
+      <Typography variant="h4" gutterBottom>
+        {match.getDisplayName()} @{" "}
+        <small>
+          {match && <Link href={`/event/${eventKey}`}>{event.name}</Link>}
+        </small>
+      </Typography>
+      <Grid container>
+        <Grid item xs={12} md={6} style={{ padding: 8 }}>
```

Comment:
The Grid container has spacing options you can use instead of `padding`

https://material-ui.com/api/grid/

Reply:
A horizontal scroll will appear if we use the ``spacing``.
https://material-ui.com/components/grid/#negative-margin